src/syscall_table/mod.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+mod aarch64;
+mod x86_64;
+
+use crate::backend::TargetArch;
+use std::collections::HashMap;
+
+/// Creates and owns a mapping from the arch-specific syscall name to the right number.
+#[derive(Debug)]
+pub(crate) struct SyscallTable {
+    map: HashMap<String, i64>,
+    arch: TargetArch,
```

Comment:
Why do we need to save the `arch`? It doesn't seem to be used internally, and it is not exported. If we need it in the future, we should just add it then instead of having an unused field.